src/models/tuning/optim.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+import logging
+import hydra
+
+from pathlib import Path
+
+from ConfigSpace import Configuration
+from smac.facade.hyperparameter_optimization_facade import (
+    HyperparameterOptimizationFacade as HPOFacade,
+)
+from smac.scenario import Scenario
+from smac.multi_objective.parego import ParEGO
+
+# from torch.cuda import is_available as cuda_available
+
+from src.models.tuning.configspace import configspace
+from src.models.train_model import train, load_data, evaluate_model
+from src.models.tuning.plot_pareto import plot_pareto
+
+# Global logger
+log_fmt = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+logging.basicConfig(level=logging.INFO, format=log_fmt)
+logger = logging.getLogger(__name__)
+
+data = load_data("./data/processed/data.csv")
+
+
+def evaluate_config(config: Configuration, seed: int = 42) -> float:
+    params = config.get_dictionary()
+
+    print("Evaluate config: " + str(params))
+
+    train_data = data.sample(frac=0.8, random_state=42)
+    test_data = data.drop(train_data.index)
+
+    model = train(train_data, params)
+
+    loss = evaluate_model(model, test_data)
+
+    print("Loss: " + str(loss.item()))
+
+    return {
+        "rmse": loss.item(),
+        "size": np.sqrt(sum(p.numel() for p in model.parameters())),
+    }
+
+
+def optimize_configuration(cfg):
```

Comment:
the optimizer configuration should also go into a .ymal - maybe we can apply our hydra knowledge here

Reply:
Thanks! Started on this, can be improved further.